### ISSUE TITLE
use prepare-binding settings for graphql prepare

### ISSUE THREAD

**Issue by marktani** [Urigo/graphql-cli#211]
```yml
projects:
  app:
    schemaPath: "src/schema.graphql"
    extensions:
      endpoints:
        default: "http://localhost:4000"
  database:
    schemaPath: "src/generated/prisma.graphql"
    extensions:
      prisma: database/prisma.yml
      prepare-binding:
        output: src/generated/prisma.ts
        generator: prisma-ts
```

---

`output` property is not picked up correctly:

```sh
graphql prepare -p database
✖ Processing schema imports for project database...
Output path cannot be determined. No existing configuration found and no output parameter specified.
```

---

If I specify the output folder using `-o`, the filename is `database.ts` and not `prisma.ts`:

```sh
graphql prepare -p database -o src/generated
✔ Bundled schema for project database written to src/generated/database.graphql
✔ Bindings for project database written to src/generated/database.ts
```
---

How can I make `graphql prepare` pick up `prepare-binding` from the configuration file?

**Comment by akoenig** [comment]
Don't want to hijack the bug report by @marktani, but would like to use the chance to propose a configuration change, because I have the feeling that the whole process of generating the files that lives in the `generated` directory is a little bit confusing / too magically. My hypothesis is: People are thinking in separate services. In fact, that is how the current architecture also looks like: The Prisma service as an own separated service and your front-facing GraphQL API which is also an own application. That is awesome and pretty much defined, but the whole architecture gets confusing if you look at the current `.graphqlconfig`. 

The process of creating the files in `generated` is:

  1. Contact the Prisma service (`database`) to fetch the schema and store it in `generated/prisma.graphql` via `graphql get-schema ...`
  2. Use the `prisma.graphql` schema and generate the respective type definitions out of it (`graphql prepare`).

Despite the fact, that the first step is not automated and has to be executed manually, you don't grasp the whole process when looking at `.graphqlconfig` IMHO. Therefore, I would like to propose the following change:

```
projects:
  app:
    schemaPath: "src/schema.graphql"
    extensions:
      endpoints:
        default: "http://localhost:4000"
      prepare-prisma-schema:
        input: "http://localhost:4466"
        output: "src/generated/prisma.graphql"
      prepare-prisma-binding:
        generator: prisma-ts
        input: "src/generated/prisma.graphql"
        output: "src/generated/prisma.ts"
  database:
    endpoints:
      default: "http://localhost:4466"
```

Basically, moving the `schema` and `binding` creation to the app project, because this is where the `generated` files also live. The whole process could be triggered via a `graphql prepare`.

Happy to hear your thoughts on this :)

**Comment by Urigo** [comment]
We've just released a new version and an alpha release channel for GraphQL CLI - `4.0.0-alpha.XXX`!

In order to receive schemas the new [new GraphQL Config](https://medium.com/the-guild/graphql-config-708dd0e5d15f) which let's you specify multiple schemas from any source possible.

I want to ask the people on this issue, in that case, why would you need the `prepare` command?
Are we missing anything and should we add this command back?

Checkout new instructions and the [migration guide](https://github.com/urigo/graphql-cli#migration-from-3xx-to-4xx) and we will be happy to hear your feedback during the alpha period!

Feel free to contact us by opening an issue or using our Discord channel;
https://discord.gg/xud7bH

**Comment by Urigo** [comment]
I've posted the wrong Discord link, this is the correct one: https://discord.gg/xud7bH9
